cmd/skaffold/app/cmd/run.go

```diff
@@ -47,7 +47,18 @@ func doRun(ctx context.Context, out io.Writer) error {
 			return fmt.Errorf("failed to build: %w", err)
 		}
 
+		if opts.SkipTests {
```

Comment:
Pretty sure this condition should be inverted.  Can you test that `--skip-tests` skips the tests? And that a failed test prevents the deploy?

Reply:
Yes :-).

Reply:
Added integration tests to cover the scenarios.